projects/hip-tests/catch/unit/kernel/hipPrintfKernel.cc

```diff
@@ -56,13 +56,11 @@ TEST_CASE("Unit_kernel_ChkPrintf") {
     if (!HipTest::isPcieAtomicSupported()) continue;
     hipLaunchKernelGGL(run_printf, dim3(1), dim3(1), 0, 0);
     HIP_CHECK(hipDeviceSynchronize());
-    char* data = new char[st.size()];
-    ;
+    char data[st.size()];
     std::ifstream CapturedData = capture.getCapturedData();
     CapturedData.getline(data, st.size() + 1);
     int result = strcmp(data, check);
     REQUIRE(result == 0);
-    delete[] data;
```

Comment:
Alternatively you could just move the delete to before the REQUIRE. This way even if the test is going to fail, it will free `data` first.